docs/source/optimizers.mdx

```diff
@@ -0,0 +1,139 @@
+## Introduction: 8-bit optimizers
+
+With 8-bit optimizers, larger models can be finetuned with the same GPU memory compared to standard 32-bit optimizer training. 8-bit optimizers are a drop-in replacement for regular optimizers:
+
+- Faster (e.g. 4x faster than regular Adam)
+- 75% less memory, same performance
+- No hyperparameter tuning needed
+
+8-bit optimizers are mostly useful to finetune large models that did not fit into memory before. They also make it easier to pretrain larger models and have great synergy with sharded data parallelism. 8-bit Adam, for example, is already used across multiple teams in Facebook. This optimizer saves a ton of memory at no accuracy hit.
+
+See here the biggest models
+
+We feature 8-bit Adam/AdamW, SGD momentum, LARS, LAMB, and RMSProp.
+
+It only requires a two-line code change to get started.
+```py
+import bitsandbytes as bnb
+
+# before: adam = torch.optim.Adam(...)
+adam = bnb.optim.Adam8bit(...)
+
+# recommended for NLP models
+# before: torch.nn.Embedding(...)
+bnb.nn.StableEmbedding(...)
+```
+
+The arguments passed are the same as standard Adam. For NLP models we recommend also to use the StableEmbedding layers which improves results and helps with stable 8-bit optimization.
+
+## Overview of supported 8-bit optimizers
+
+TOOD: List here all optimizers in `bitsandbytes/optim/__init__.py`
+TODO (future) document all optimizers as we did for Linear4bit / Linear8bitLt classes
+
+## Overview of expected gradients
```

Comment:
What this means exactly? Can you elaborate more about "expected gradients"? I also thing this is too in-depth and should be moved below